src/Storages/IStorage_fwd.h

```diff
@@ -12,6 +12,7 @@ class IStorage;
 
 using ConstStoragePtr = std::shared_ptr<const IStorage>;
 using StoragePtr = std::shared_ptr<IStorage>;
+using StorageWeakPtr = std::weak_ptr<IStorage>;
```

Comment:
It's not allowed to use `std::weak_ptr<IStorage>` because we rely on `std::shared_ptr<IStorage>::unique()`

Reply:
Instead of saving StoragePtr, we can save StorageID with resolved UUID

Reply:
it's exactly the point to not keep shared_ptr here - since on HTTP requests session can be preserved it would block database from DROP - making it weak_ptr doesn't block it and we don't need to keep it cached beyond session